docker-compose.yml

```diff
@@ -351,7 +353,65 @@ services:
       - .:/arrow:delegated
       - ${DOCKER_VOLUME_PREFIX}ubuntu-ccache:/ccache:delegated
     command: *cpp-command
+  
+  ubuntu-cpp-static:
+    # Usage:
+    #   docker-compose build ubuntu-cpp-static
+    #   docker-compose run --rm ubuntu-cpp-static
+    # Parameters:
+    #   ARCH: amd64, arm64v8, s390x, ...
+    #   UBUNTU: 18.04, 20.04
+    image: ${REPO}:${ARCH}-ubuntu-${UBUNTU}-cpp-static
+    build:
+      context: .
+      dockerfile: ci/docker/ubuntu-${UBUNTU}-cpp.dockerfile
+      cache_from:
+        - ${REPO}:${ARCH}-ubuntu-${UBUNTU}-cpp-static
+      args:
+        arch: ${ARCH}
+        base: "${ARCH}/ubuntu:${UBUNTU}"
+        clang_tools: ${CLANG_TOOLS}
+        llvm: ${LLVM}
+        gcc_version: ${GCC_VERSION}
+    shm_size: *shm-size
+    cap_add:
+      - SYS_ADMIN
+    devices:
+      - "/dev/fuse:/dev/fuse"
+    security_opt:
+      - apparmor:unconfined
+    ulimits: *ulimits
+    environment:
+      <<: *ccache
+      ARROW_HOME: /arrow
+      ARROW_DEPENDENCY_SOURCE: BUNDLED
```

Comment:
I can't see anything here which indicates a static build.

Reply:
We define static builds in `tasks.yml` like the following:

```
  test-ubuntu-18.04-cpp-static:
    ci: github
    template: docker-tests/github.linux.yml
    params:
      env:
        UBUNTU: 18.04
      flags: "-e ARROW_BUILD_SHARED=OFF -e ARROW_BUILD_STATIC=ON -e ARROW_TEST_LINKAGE=static"
      image: ubuntu-cpp
```

Reply:
I named them like this as they run `r/inst/build_arrow_static.sh` which does set `ARROW_BUILD_STATIC=ON`. But we can rename them. Maybe `*-cpp-r-static` or something like you suggested in another comment?

As this is part of moving over the builds from https://github.com/ursa-labs/arrow-r-nightly I first wanted to move the jobs over as 1:1 as possible and optimize in a follow up.